### PR TITLE
[X] Add compiled converters

### LINKED ISSUE #6547: Provide a compiled version of FlexEnums and FlowDirection
### Description

we have [TypeConverter] on some enums, allowing slightly different syntaxes for some reasons (CSS parsing), but adding those mark the value ineligible for XamlCompilation.

we should provide a compiled version of all of those, even if it's a special kind of converter that revert to enum parsing

### Public API Changes

/

### Intended Use-Case

/

### PR DESCRIPTION
**Pull request by StephaneDelcroix**
### Description of Change

- Add compiled converters for Flex enums
- and for FlexBasis
- and for for FlowDirection
- Also moves Flowdirection converter attribute from the property to the
  Type, move the converter from Controls to Core, and fix the Design assembly accordingly

### Issues Fixed

- fixes #6547

### CODE REVIEW

File: src/Core/src/Converters/FlexEnumsConverters.cs [hunk 1]
```diff
@@ -218,9 +218,9 @@ public override object ConvertFrom(ITypeDescriptorContext context, CultureInfo c
 			var strValue = value?.ToString();
 			if (strValue != null)
 			{
+				value = strValue.Trim();
```
Comment (campersau) [inline src/Core/src/Converters/FlexEnumsConverters.cs:221]: `value` does not seem to be used later. Maybe trim the `strValue`?

```suggestion
```

Comment (StephaneDelcroix) [reply src/Core/src/Converters/FlexEnumsConverters.cs:221]: 👍

File: src/Controls/src/Build.Tasks/CompiledConverters/FlexBasisTypeConverter.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+{
+	class FlexBasisTypeConverter : ICompiledTypeConverter
+	{
+		public IEnumerable<Instruction> ConvertFromString(string value, ILContext context, BaseNode node)
+		{
+			var module = context.Body.Method.Module;
+			if (!string.IsNullOrEmpty(value))
+			{
+				value = value.Trim();
+				if (value == "Auto")
```
Comment (mattleibow) [inline src/Controls/src/Build.Tasks/CompiledConverters/FlexBasisTypeConverter.cs:19]: Is this case sensitive?

**Review by mattleibow — APPROVED** [review]

File: src/Core/src/Layouts/FlexEnums.cs [hunk 3]
```diff
@@ -75,10 +75,12 @@ public struct FlexBasis
 	{
 		bool _isLength;
 		bool _isRelative;
-		public static FlexBasis Auto = new();
 		public float Length { get; }
 		internal bool IsAuto => !_isLength && !_isRelative;
 		internal bool IsRelative => _isRelative;
+
+		public static FlexBasis Auto = new();
```
[the remark this answers is not in the document]
Comment (StephaneDelcroix) [reply src/Core/src/Layouts/FlexEnums.cs:82]: won't change a thing, but doesn't hurt

**Pull request merged**